Source/WebCore/platform/network/soup/ResourceRequestSoup.cpp

```diff
@@ -140,6 +140,24 @@ void ResourceRequest::updateSoupMessageBody(SoupMessage* soupMessage, BlobRegist
 
 }
 
+GRefPtr<GBytes> ResourceRequest::createBodyBytes() const
+{
+    auto formData = httpBody();
+    if (!formData || formData->isEmpty())
+        return nullptr;
+
+    auto resolvedFormData = formData->resolveBlobReferences();
+    uint64_t length = 0;
+    for (auto& element : resolvedFormData->elements())
+        length += element.lengthInBytes();
+
+    if (!length)
+        return nullptr;
+
+    auto bytes = resolvedFormData->flatten();
+    return adoptGRef(g_bytes_new_static(bytes.data(), bytes.size()));
```

Comment:
Ah, I was about to land this but realized this can actually be simplified further. There's no need to loop through resolvedFormData:

```suggestion
    auto bytes = formData->resolveBlobReferences()->flatten();
    if (!bytes.size())
        return nullptr;
    return adoptGRef(g_bytes_new_static(bytes.data(), bytes.size()));
```

Reply:
Another option would be even simpler:

```suggestion
    auto bytes = formData->resolveBlobReferences()->flatten();
    return adoptGRef(g_bytes_new_static(bytes.data(), bytes.size()));
```

In this case, the public API return value would be non-nullable, but the GBytes would be empty and contain NULL data of 0 size. Maybe this would be better so that callers don't have to NULL-check whether the GBytes itself is NULL? The caller would only need to check the data that it contains.